src/rpc/request.rs

```diff
@@ -704,9 +704,66 @@ pub enum Request {
     #[api(type = 1313)]
     #[display("address_list({0})")]
     AddressList(List<bitcoin::Address>),
+
+    #[api(type = 1314)]
+    #[display("set_offer_history({0})")]
+    SetOfferStatus(OfferStatusPair),
+
+    #[api(type = 1315)]
+    #[display("retrieve_offers")]
+    RetrieveOffers(OfferStatusSelector),
+
+    #[api(type = 1316)]
+    #[display("offer_status_list({0})")]
+    OfferStatusList(List<OfferStatusPair>),
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode)]
+#[derive(Clone, Debug, Eq, PartialEq, Display, StrictEncode, StrictDecode)]
+#[display("{offer}, {status}")]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+#[display(OfferStatusPair::to_yaml_string)]
+pub struct OfferStatusPair {
+    pub offer: PublicOffer<BtcXmr>,
+    pub status: OfferStatus,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Display, StrictEncode, StrictDecode)]
+pub enum OfferStatusSelector {
+    #[display("Open")]
+    Open,
+    #[display("In Progress")]
+    InProgress,
+    #[display("Ended")]
+    Ended,
+    #[display("All")]
+    All,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Display, StrictEncode, StrictDecode)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+pub enum OfferStatus {
+    #[display("Open")]
+    Open,
+    #[display("In Progress")]
+    InProgress,
+    #[display("Ended({0})")]
+    Ended(Outcome),
+}
+
```

Comment:
`Expired`?

Reply:
`Expired` has its own semantics for me, I'd rather think of it as a variant of the `Outcome` enum.

Reply:
I see. For me `Outcome` is about a swap, not an offer, and an offer that was never taken never becomes an `Outcome`, its just an expired offer

Reply:
For me a `Trade` is a swap, not an `Offer`, and only a trade has an `Outcome`, so I don't see how that helps

Reply:
But it doesnt matter for now anyway

Reply:
`Trade` would be different than `Offer`: this should allow use to have a proper `Offer` type (with room for negotiation) and a `Trade` type (close to what we currently have in `Offer`) that represent all the parameters of a swap.

`Offer` is kind of in between currently.